PROTO_tests/tests/dtypes_test.py

```diff
@@ -0,0 +1,168 @@
+    def test_pdarrays_datatypes(self, size):
+        assert dtypes.dtype("int64") == ak.array(np.arange(size)).dtype
+        assert dtypes.dtype("uint64") == ak.array(np.arange(size), ak.uint64).dtype
+        assert dtypes.dtype("bool") == ak.ones(size, ak.bool).dtype
+        assert dtypes.dtype("float64") == ak.ones(size).dtype
+        assert dtypes.dtype("str") == ak.random_strings_uniform(1, 16, size=size).dtype
+
+        bi = ak.bigint_from_uint_arrays(
+            [ak.ones(size, dtype=ak.uint64), ak.arange(size, dtype=ak.uint64)]
+        ).dtype
+        assert dtypes.dtype("bigint") == bi
+        assert dtypes.dtype("bigint") == ak.arange(2 ** 200, 2 ** 200 + size).dtype
+
+    def test_isSupportedInt(self):
+        for supported in -10, 1, np.int64(1), np.int64(1.0), np.uint32(1), 2 ** 63 + 1, 2 ** 200:
+            assert dtypes.isSupportedInt(supported)
+        for unsupported in 1.0, "1":
+            assert not dtypes.isSupportedInt(unsupported)
+
+    def test_isSupportedFloat(self):
+        for supported in np.nan, -np.inf, 3.1, -0.0, float(1), np.float64(1):
+            assert dtypes.isSupportedFloat(supported)
+        for unsupported in np.int64(1.0), int(1.0), "1.0":
+            assert not dtypes.isSupportedFloat(unsupported)
+
+    def test_DtypeEnum(self):
+        assert "bool" == str(dtypes.DType.BOOL)
+        assert "float" == str(dtypes.DType.FLOAT)
+        assert "float64" == str(dtypes.DType.FLOAT64)
+        assert "int" == str(dtypes.DType.INT)
+        assert "int64" == str(dtypes.DType.INT64)
+        assert "str" == str(dtypes.DType.STR)
+        assert "uint8" == str(dtypes.DType.UINT8)
+        assert "bigint" == str(dtypes.DType.BIGINT)
+
+        enum_vals = frozenset(
+            {"float", "float64", "bool", "uint8", "int", "int64", "str", "uint64", "bigint"}
+        )
+        assert enum_vals == ak.DTypes
+        assert enum_vals == ak.ARKOUDA_SUPPORTED_DTYPES
+
+    def test_NumericDTypes(self):
+        num_types = frozenset(["bool", "float", "float64", "int", "int64", "uint64", "bigint"])
+        assert num_types == dtypes.NumericDTypes
+
+    def test_SeriesDTypes(self):
+        assert np.str_ == dtypes.SeriesDTypes["string"]
+        assert np.str_ == dtypes.SeriesDTypes["<class 'str'>"]
+        assert np.int64 == dtypes.SeriesDTypes["int64"]
+        assert np.int64 == dtypes.SeriesDTypes["<class 'numpy.int64'>"]
+        assert np.float64 == dtypes.SeriesDTypes["float64"]
+        assert np.float64 == dtypes.SeriesDTypes["<class 'numpy.float64'>"]
+        assert np.bool_ == dtypes.SeriesDTypes["bool"]
+        assert np.dtype(bool) == dtypes.SeriesDTypes["bool"]
+        assert np.bool_ == dtypes.SeriesDTypes["<class 'bool'>"]
+        assert np.dtype(bool) == dtypes.SeriesDTypes["<class 'bool'>"]
+        assert np.int64 == dtypes.SeriesDTypes["datetime64[ns]"]
+        assert np.int64 == dtypes.SeriesDTypes["timedelta64[ns]"]
```

Comment:
This can be simplified / cleaned up
```python
def test_SeriesDTypes(self):
  for dt in "int64", "<class 'numpy.int64'>", "datetime64[ns]", "timedelta64[ns]":
    assert dtypes.SeriesDTypes[dt] == np.int64
  
  for dt in "string", "<class 'str'>":
    assert dtypes.SeriesDTypes[dt] == np.str_
  
  for dt in "float64", "<class 'numpy.float64'>":
    assert dtypes.SeriesDTypes[dt] == np.float64
  
  for dt in "bool", "<class 'bool'>":
    assert dtypes.SeriesDTypes[dt] == np.bool_
```

I removed the checks for `np.dtype(bool)` because this is equivalent to `np.bool_`
```python
In [3]: np.dtype(bool) == np.bool_
Out[3]: True
```

Reply:
Updated